ui/packages/catalog-ui-search/src/main/webapp/react-component/filter/filter.js

```diff
@@ -0,0 +1,370 @@
+const BetweenTimeView = require('../../component/between-time/between-time.view.js')
+const ValueModel = require('../../component/value/value.js')
+const properties = require('../../js/properties.js')
+const Common = require('../../js/Common.js')
+import * as React from 'react'
+import {
+  geometryComparators,
+  dateComparators,
+  stringComparators,
+  numberComparators,
+  booleanComparators,
+} from '../../component/filter/comparators'
+import ExtensionPoints from '../../extension-points'
+import withListenTo from '../../react-component/container/backbone-container'
+import MarionetteRegionContainer from '../../react-component/container/marionette-region-container'
+import styled from 'styled-components'
+
+const generatePropertyJSON = (value, type, comparator) => {
+  const propertyJSON = _.extend({}, metacardDefinitions.metacardTypes[type], {
+    value,
+    multivalued: false,
+    enumFiltering: true,
+    enumCustom: true,
+    matchcase: ['MATCHCASE', '='].indexOf(comparator) !== -1 ? true : false,
+    enum: metacardDefinitions.enums[type],
+    showValidationIssues: false,
+  })
+
+  if (propertyJSON.type === 'GEOMETRY') {
+    propertyJSON.type = 'LOCATION'
+  }
+
+  if (propertyJSON.type === 'STRING') {
+    propertyJSON.placeholder = 'Use * for wildcard.'
+  }
+
+  if (comparator === 'NEAR' && propertyJSON.type === 'STRING') {
+    propertyJSON.type = 'NEAR'
+    propertyJSON.param = 'within'
+    propertyJSON.help =
+      'The distance (number of words) within which search terms must be found in order to match'
+    delete propertyJSON.enum
+  }
+
+  // if we don't set this the property model will transform the value as if it's a date, clobbering the special format
+  if (
+    comparator === 'RELATIVE' ||
+    (comparator === 'BETWEEN' && propertyJSON.type === 'DATE')
+  ) {
+    propertyJSON.transformValue = false
+  }
+
+  return propertyJSON
+}
+
+const determineView = comparator => {
+  let necessaryView
+  switch (comparator) {
+    case 'RELATIVE':
+      necessaryView = RelativeTimeView
+      break
+    case 'BETWEEN':
+      necessaryView = BetweenTimeView
+      break
+    default:
+      necessaryView = MultivalueView
+      break
+  }
+  return necessaryView
+}
+
+const transformValue = (value, comparator) => {
+  switch (comparator) {
+    case 'NEAR':
+      if (value[0].constructor !== Object) {
+        value[0] = {
+          value: value[0],
+          distance: 2,
+        }
+      }
+      break
+    case 'INTERSECTS':
+    case 'DWITHIN':
+      break
+    default:
+      if (value === null || value[0] === null) {
+        value = ['']
+        break
+      }
+      if (value[0].constructor === Object) {
+        value[0] = value[0].value
+      }
+      break
+  }
+  return value
+}
+
+const FilterInput = styled.div`
+    ${({ type, theme }) =>
+      type === 'DATE'
+        ? `min-width: calc(19 * ${theme.minimumFontSize}) !important;`
```

Comment:
Hayley used `19 * ${theme.minimumFontSize}` in her most recent DDF PR as well. Here: https://github.com/codice/ddf/pull/5118/files#diff-dc1646bfebd9c7a0d44306795a1660a1R68 , and other places. 
Should we extract that and put it somewhere that it can be shared?

Reply:
I think I'll have to rebase and fix merge conflicts once hers gets merged